nabs/plans.py

```diff
@@ -699,26 +699,43 @@ def fixed_target_scan(detectors, x_motor, xx, y_motor, yy, scan_motor, ss,
     """
 
     detectors = list(detectors)
-    # detectors = [daq, sequencer]
-
+    # detectors = [daq, seq]
+    if n_targets and n_targets > len(xx):
+        msg = ('The number of samples: %s is bigger then the available'
+               ' samples: %s. Please provide a number in range.',
+               n_targets, len(xx))
+        logger.warning(msg)
+        raise IndexError(msg)
+
+    # TODO: check if xx, yy divisible by n_targets?? or handle if not....
+    # # n_targets == n_shots
     def inner_scan():
-        # starting from the beginning of the list.
-        next_target = 0
-        for i in range(n1):
-            #  scan_motor.move_to_step(N)
-            yield from bp.list_scan([], scan_motor, [ss[i]])
-            for j in range(n2):
-                # xy.moveToNextTarget()
-                yield from bp.list_scan([], x_motor, [xx[next_target]],
-                                        y_motor, [yy[next_target]])
-                next_target += 1
-                # pp.get_burst(1)
-                yield Msg('open_run')
-                yield from bps.trigger_and_read(detectors + [x_motor,
-                                                y_motor, scan_motor])
-                yield Msg('close_run')
+        for j in range(n1):
+            yield from bp.list_scan([], scan_motor, [ss[j]])
+            x_pos = xx[(j * n_targets):((j + 1) * n_targets)]
+            y_pos = yy[(j * n_targets):((j + 1) * n_targets)]
+            yield from bp.list_scan(detectors, x_motor, x_pos, y_motor, y_pos)
```

Comment:
Assuming I have `detectors = [seq]` here, the list_scan should do a `trigger` message correct? If this is so, then it will send a trigger to the sequencer and that should call the specific sequence (pp.burst sequence)???

Or do I specifically do a `yield from trigger(detectors)` ?

Or do I have to actually have the pulse picker as a parameter to the scan???

What about the pulse_picker being "optional"? In case the users want to use the sequencer but not the pulse picker, do they have to change the sequence and run the scan again???

Reply:
Also do I have to have the second function specific with `daq` or can I just add the `@nbpp.daq_during_decorator(record=record, controls=control_devices)` decorator here for the `inner_scan` like @ZryletTC  mentioned in the chat?

Or do we make it separate because we don't always use `daq`?

Reply:
@silkenelson I would like some clarifications about the scan if you could help me - this should be something simple but it is taking me days... 

Should I assume in my scan that I will send an `pp` object, so I can do something like this:
```
if use_pp:
     pp.get_burst(1)
     pp.prepare_burst(xxx) 
```
Or should I assume that i will only have a sequencer, and the sequencer will be running the correct sequence (including the one set for pp).

Reply:
The `list_scan` takes care of triggering and reading for you.
How do you figure that out?
`list_scan` yields from `scan_nd`:
https://github.com/bluesky/bluesky/blob/6e799a77db42ad83eebbeb0aa6e24bee24fe99d0/bluesky/plans.py#L176-L177
`scan_nd`, at each step, by default calls `one_nd_step`:
https://github.com/bluesky/bluesky/blob/6e799a77db42ad83eebbeb0aa6e24bee24fe99d0/bluesky/plan_stubs.py#L1087-L1113

`one_nd_step` `take_reading` by default is `trigger_and_read`. `trigger_and_read` is:
https://github.com/bluesky/bluesky/blob/6e799a77db42ad83eebbeb0aa6e24bee24fe99d0/bluesky/plan_stubs.py#L869-L912

Which, you can see, yields 'trigger' and 'read' messages as the name implies. 

Without digging into the source above, you can inspect plans to see what bluesky is doing (https://blueskyproject.io/bluesky/simulation.html).  bluesky's documentation is pretty good, so referring back to it is usually worthwhile.

Or if you're feeling particularly hacky, you could always make a device with a `trigger` method and `print('triggered!')` yourself to see what `list_scan` does to your detectors.

It's a bit tough to read through on the first go, but it should get easier as you get more familiar with it.